include/openthread/dataset.h

```diff
@@ -127,6 +133,12 @@ struct otPskc
  */
 typedef struct otPskc otPskc;
 
+/**
+ * This datatype represents KeyRef to PSKc.
+ *
+ */
+typedef uint32_t otPskcRef; ///< Reference to Key
```

Comment:
We have different types `otNetworkKeyRef` and `otPskcRef`, ect (and all are defined as `uint32_t`).

Do we need to distinguish between them? Can we define and use the common `otCryptoKeyRef` for all key refs?

(checking the code I think they all end up using `otPlatCryptoImportKey()` and `otPlatCryptoExportKey()` which uses `otCryproKeyRef`).

Reply:
We can, This was a legacy decision in the initial PR. And based on previous comments I separated out the NetwokKey and Pskc Refs.

Reply:
@abtink I think I prefer having `otNetworkKeyRef` and `otPskcRef` separated instead of using `otCryproKeyRef` as it makes more intuitive sense. The user can easily see the new datatypes to use. Using `otCryproKeyRef` is far less intuitive for a developer. It won't have any memory penalty as well. What do you think?

Reply:
The advantage I see by using `otCryptoKeyRef` for all is that it re-enforces the fact they are all going to be used with same platform model (i.e. there is PSA storage and platform APIs where we can import and export the keys from by knowing this key ref). And if we need to change the `otCryptoKeyRef` later (maybe it needs to be `strcut`  or different `uint` length) then we don't need to change multiple place, just changing `otCryptoKeyRef` would suffice.

If you want to keep different definitions for `otPskRef` and `otNetworkKeyRef` I suggest we `typedef` them as `otCryptoKeyRef`, i.e.,
```c++
typedef otCryptoKeyRef otPskcRef;
```